Don't depend on system krb5.conf in test suite

frozencemetery opened the pull request:
It's my own fault that I ended up caring about this behavior, but this may be more broadly useful in case anyone hits it again.  Anyway, this fixes tests which break if there are problems parsing /etc/krb5.conf from the system.

greghudson commented:
I agree that tests should not access /etc/krb5.conf.

The idiom you used is taken from lib/krb5/krb/Makefile.in, which actually has a t_krb5.conf file, but is used in directories which do not.  An empty krb5.conf probably does not cause any trouble for the tests, but it's weird to use such a specific pathname for a file which does not exist.  Also, I think there are less verbose solutions available; "KRB5_CONF=pathname command ..." could be used instead of explicitly exporting KRB5_CONF, and a make variable could be used to avoid the repetition.

frozencemetery commented:
Oh, that makes sense.  And it probably looked like it worked okay because empty is superior to broken...

How would you prefer that the cases where t_krb5.conf doesn't exist be handled?  `KRB5_CONF=/dev/null` or some such?

greghudson commented:
/dev/null is probably reasonable.  Even on platforms where it doesn't exist, it won't affect the tests.

frozencemetery commented:
New version hopefully addressing comments.  The expansion of `RUN_SETUP` uses the `VARIABLE=value; export $VARIABLE` idiom, so setting `KRB5_CONFIG` without using this idiom has to occur after `RUN_SETUP`.

greghudson commented:
Please see PR #341; I decided to make some deeper build system changes to make running C test programs from Makefiles cleaner.
